presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -195,6 +195,10 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
 
         Set<QualifiedTablePrefix> prefixes = getPrefixes(session, table, constraint);
 
+        if (prefixes.equals(defaultPrefixes())) {
```

Comment:
`if (prefixes.equals(table.getPrefixes())`?

Should we remove `|| !table.getPrefixes().equals(defaultPrefixes())` from line 192?

Reply:
I moved the checking of `isTablesEnumeratingTable` from `calculatePrefixesWithSchemaName` to line 192 to short-circuit it. As there is no `isTablesEnumeratingTable` check in `calculatePrefixesWithSchemaName` any more, we can't remove `|| !table.getPrefixes().equals(defaultPrefixes())` from line 192.

Reply:
Sorry I misread your suggestion and my reasoning was messed up.

I see your point now.
1. For the first suggestion, yes, I think it should be `if (prefixes.equals(table.getPrefixes())`
2. For the second suggestion, I'd rather keep `|| !table.getPrefixes().equals(defaultPrefixes())` for now because the current `applyFilter` doesn't perform predicate pushdown upon existing prefixes. (Each time prefixes are calculated independently) We can address this in a future PR.